app/controllers/gift_requests_controller.rb

```diff
@@ -1,7 +1,12 @@
 class GiftRequestsController < ApplicationController
+  before_filter :authenticate_user!
+  before_action only: [:new, :edit, :create, :update, :destroy_multiple] do
+    head :forbidden if !current_user.therapist?
```

Comment:
Always favor redirecting over rendering a status code - a status code is a full stop to a user's experience.

It'd be sensible to just put them on the root page.